system/jlib/jutil.cpp

```diff
@@ -3056,21 +3056,37 @@ extern jlib_decl void setSecretMount(const char * path)
         secretDirectory.set(path);
 }
 
-extern jlib_decl StringBuffer & getSecret(StringBuffer & result, const char * name, const char * key)
+extern jlib_decl StringBuffer & getSecret(StringBuffer & result, const char * name, const char * key, bool required)
```

Comment:
Maybe we should also export a

    extern jlib_decl IFile * getSecretFile(const char * name, const char * key);

that these functions can use, and can be used externally.

Reply:
although that would tie us to using a mount as an implementation, so I don't think it should be public.  We may want to extend the implementation to use other vault implementations.,  In which case there will not be an underlying file.  At that point we should probably introduce an ISecret though to avoid checks for existance followed by retrieving the value.

Reply:
I just directly violated the idea that they aren't mounted with my tls change.  I hadn't read your comment yet.  I'll keep looking at the tls stuff and try to get rid of needing to use a file rather then memory.  I've gotten further but still haven't quite put a good test together.